std/os/uefi/protocols/absolute_pointer_protocol.zig

```diff
@@ -0,0 +1,64 @@
+const uefi = @import("std").os.uefi;
+const Event = uefi.Event;
+const Guid = uefi.Guid;
+
+/// UEFI Specification, Version 2.8, 12.7
+pub const AbsolutePointerProtocol = extern struct {
+    _reset: extern fn (*const AbsolutePointerProtocol, bool) usize,
+    _get_state: extern fn (*const AbsolutePointerProtocol, *AbsolutePointerState) usize,
+    wait_for_input: *Event,
+    mode: *AbsolutePointerMode,
+
+    pub fn reset(self: *const AbsolutePointerProtocol, verify: bool) usize {
+        return self._reset(self, verify);
+    }
+
+    pub fn getState(self: *const AbsolutePointerProtocol, state: *AbsolutePointerState) usize {
+        return self._get_state(self, state);
+    }
+
+    pub const guid align(8) = Guid{
+        .time_low = 0x8d59d32b,
+        .time_mid = 0xc655,
+        .time_high_and_version = 0x4ae9,
+        .clock_seq_high_and_reserved = 0x9b,
+        .clock_seq_low = 0x15,
+        .node = [_]u8{ 0xf2, 0x59, 0x04, 0x99, 0x2a, 0x43 },
+    };
+};
+
+pub const AbsolutePointerMode = extern struct {
+    absolute_min_x: u64,
+    absolute_min_y: u64,
+    absolute_min_z: u64,
+    absolute_max_x: u64,
+    absolute_max_y: u64,
+    absolute_max_z: u64,
+    attributes: packed struct {
+        _pad1: u6,
+        supports_pressure_as_z: bool,
+        supports_alt_active: bool,
+        _pad2: u24,
+    },
+};
+
+pub const AbsolutePointerState = extern struct {
+    current_x: u64,
+    current_y: u64,
+    current_z: u64,
+    active_buttons: packed struct {
+        _pad1: u6,
+        alt_active: bool,
+        touch_active: bool,
+        _pad2: u24,
+    },
+
+    pub fn init() AbsolutePointerState {
+        return AbsolutePointerState{
+            .current_x = undefined,
```

Comment:
is this a TODO? why would someone use this `init` function over just `var x:AbsolutePointerState = undefined;`

Reply:
I wrote that when I didn't know default values have been implemented.

Reply:
Why would you ever have the default values set to `undefined` though?

Reply:
AbsolutePointerState contains information about where the screen was touched. The structure is passed to and filled with information by getState. As far as I know that's the only thing this structure is good for. We could pre-fill it with specific values of our own, but I don't see how that would make any sense. Anything meaningful inside this struct is returned from UEFI, never passed to it. Instead of having any defaults set, we could assign something everytime we create an AbsolutePointerState, but it would be overwritten as soon as we use it with getState and the code would become bloated.